pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -351,8 +351,15 @@ func translateRoute(in *networking.HTTPRoute,
 		for _, dst := range in.Route {
 			weight := &types.UInt32Value{Value: uint32(dst.Weight)}
 			if dst.Weight == 0 {
-				weight.Value = uint32(100)
+				// Ignore 0 weighted clusters if there are other clusters in the route.
+				// But if this is the only cluster in the route, then add it as a cluster with weight 100
+				if len(in.Route) == 1 {
+					weight.Value = uint32(100)
+				} else {
+					continue
```

Comment:
That would be too much. This is a common use case:
```yaml
route:
- destination:
    name: foo
   weight: 0
- destination:
     name: bar
   weight: 90
- destination:
     name: scooby
   weight: 10
```

and a log line saying weights ignored will cause more alarm don't you think? if people don't understand the logic properly

Reply:
What I'm thinking is add an info level log, otherwise, people may confused why my weight does not take effect. Sometimes they may not intend to configure it as 0, but configured as 0 by mistake. Comments?

Reply:
if weight is not set, it defaults to 100. if weight is intentionally set to 0, then it is what it is.. I am adding this support based on what a user actually asked for in the mailing list (needing to set 0 weights).

Reply:
Ah, I see, then I think we should also update document to clarify this behavior after this PR merged.